src/components/utils/popover.ts

```diff
@@ -0,0 +1,352 @@
+import SearchInput from './search-input';
+import EventsDispatcher from './events';
+import { isMobile } from '../utils';
+
+/**
+ * Describe parameters for rendering the single item of Popover
+ */
+export interface PopoverItem {
+  /**
+   * Item icon to be appeared near a title
+   */
+  icon: string;
+
+  /**
+   * Displayed text
+   */
+  label: string;
+
+  /**
+   * Item name
+   * Used in data attributes needed for cypress tests
+   */
+  name?: string;
+
+  /**
+   * Additional displayed text
+   */
+  secondaryLabel?: string;
+
+  /**
+   * Itm click handler
+   *
+   * @param item - clicked item
+   */
+  onClick: (item: PopoverItem) => void;
+}
+
+/**
+ * Event that can be triggered by the Popover
+ */
+export enum PopoverEvent {
+  /**
+   * When popover overlay is clicked
+   */
+  OverlayClicked = 'overlay-clicked',
+}
+
+/**
+ * Popover is the UI element for displaying vertical lists
+ */
+export default class Popover extends EventsDispatcher<PopoverEvent> {
+  /**
+   * Items list to be displayed
+   */
+  private readonly items: PopoverItem[];
+
+  /**
+   * Created nodes
+   */
+  private nodes: {
+    wrapper: HTMLElement;
+    popover: HTMLElement;
+    items: HTMLElement;
+    nothingFound: HTMLElement;
+    overlay: HTMLElement;
+  } = {
+    wrapper: null,
+    popover: null,
+    items: null,
+    nothingFound: null,
+    overlay: null,
+  }
+
+  /**
+   * Additional wrapper's class name
+   */
+  private readonly className: string;
+
+  /**
+   * Listeners util instance
+   */
+  private listeners: Listeners;
+
+  /**
+   * Flipper - module for keyboard iteration between elements
+   */
+  private flipper: Flipper;
+
+  /**
+   * Pass true to enable local search field
+   */
+  private searchable: boolean;
+  private search: SearchInput;
+  private filterLabel: string;
+  private nothingFoundLabel: string;
+
+  /**
+   * Style classes
+   */
+  private static get CSS(): {
+    popover: string;
+    popoverOpened: string;
+    itemsWrapper: string;
+    item: string;
+    itemHidden: string;
+    itemFocused: string;
+    itemLabel: string;
+    itemIcon: string;
+    itemSecondaryLabel: string;
+    noFoundMessage: string;
+    noFoundMessageShown: string;
+    popoverOverlay: string;
+    popoverOverlayHidden: string;
+    documentScrollLocked: string;
+    } {
+    return {
+      popover: 'ce-popover',
+      popoverOpened: 'ce-popover--opened',
+      itemsWrapper: 'ce-popover__items',
+      item: 'ce-popover__item',
+      itemHidden: 'ce-popover__item--hidden',
+      itemFocused: 'ce-popover__item--focused',
+      itemLabel: 'ce-popover__item-label',
+      itemIcon: 'ce-popover__item-icon',
+      itemSecondaryLabel: 'ce-popover__item-secondary-label',
+      noFoundMessage: 'ce-popover__no-found',
+      noFoundMessageShown: 'ce-popover__no-found--shown',
+      popoverOverlay: 'ce-popover__overlay',
+      popoverOverlayHidden: 'ce-popover__overlay--hidden',
+      documentScrollLocked: 'ce-scroll-locked',
```

Comment:
We should really think of using css modules

Reply:
We will, at some of the next udpates